qiskit/primitives/containers/data_bin.py

```diff
@@ -54,10 +63,33 @@ def __repr__(self):
         vals = (f"{name}={getattr(self, name)}" for name in self._FIELDS if hasattr(self, name))
         return f"{type(self)}({', '.join(vals)})"
 
+    def __getitem__(self, key: str) -> Any:
+        if not hasattr(self, key):
+            raise KeyError(f"Key ({key}) does not exist in this data bin.")
+        return getattr(self, key)
+
+    def __contains__(self, key: str) -> bool:
+        return hasattr(self, key)
+
+    def __iter__(self) -> Iterable[str]:
+        return iter(self._FIELDS)
+
+    def keys(self) -> list[str]:
+        """Return a list of field names."""
+        return self._FIELDS
+
+    def values(self) -> list[Any]:
```

Comment:
```suggestion
    def values(self) -> tuple[Any]:
```
I'm assuming we want this to mimic an immutable mapping, rather than a mutable one, so I think this should be an immutable type

Reply:
Also im not sure if this is possible to implement,, but technically we know the actual types of all the entries in this tuple in a specific databin class

Reply:
The actual types are `_FIELD_TYPES`. I'm not sure how I use it as a type hint. Do you have any suggestion?